src/App.tsx

```diff
@@ -15,48 +16,104 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortType {
+  LENGTH = 'length',
+  ALPHABET = 'alphabet',
+  DEFAULT = '',
+}
+
+interface SortParams {
+  sortField: SortType;
+  isReversed: boolean;
+}
+
+function getPreparedGoods(goods: string[],
+  { sortField, isReversed }: SortParams): string[] {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    switch (sortField) {
+      case SortType.ALPHABET:
+        preparedGoods.sort((good1, good2) => good1.localeCompare(good2));
+        break;
+      case SortType.LENGTH:
+        preparedGoods.sort((good1, good2) => good1.length - good2.length);
+        break;
+      default:
+        return preparedGoods;
+    }
+  }
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [isReversed, setReversed] = useState(false);
+  const [sortField, setSortField] = useState(SortType.DEFAULT);
+
+  const visibleGoods = getPreparedGoods(goodsFromServer,
+    { sortField, isReversed });
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={classNames('button', 'is-info', {
+            'is-light': sortField !== SortType.ALPHABET,
+          })}
+          onClick={() => {
+            setSortField(SortType.ALPHABET);
+          }}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={classNames('button', 'is-success', {
+            'is-light': sortField !== SortType.LENGTH,
+          })}
+          onClick={() => {
+            setSortField(SortType.LENGTH);
+          }}
         >
           Sort by length
         </button>
-
         <button
           type="button"
-          className="button is-warning is-light"
+          className={classNames('button', 'is-warning', {
+            'is-light': isReversed === false,
+          })}
+          onClick={() => {
+            setReversed(!isReversed);
+          }}
```

Comment:
move this handler to an own function
```
const handleReverseClick = () => {
  setReversed(prev => !prev);
}
```

```suggestion
          onClick={handleReverseClick}
```